crates/shim-sgx/src/main.rs

```diff
@@ -249,15 +250,26 @@ pub unsafe extern "sysv64" fn _start() -> ! {
 }
 
 unsafe extern "C" fn main(
-    port: &mut [usize; BLOCK_SIZE / core::mem::size_of::<usize>()],
+    block: &mut [usize; BLOCK_SIZE / core::mem::size_of::<usize>()],
     ssas: &mut [StateSaveArea; 3],
     cssa: usize,
 ) {
     ssas[cssa].extra[0] = 1; // Enable exceptions
+                             //
+    let block_start = block.as_ptr() as usize;
+    let block_end = block_start + BLOCK_SIZE;
+    let shim_start = shim_address();
+    let shim_end = shim_start + ENCL_SIZE;
+
+    if (block_start >= shim_start && block_start < shim_end)
```

Comment:
Let's leave a comment here explaining why it's important for security to verify the block like this.

Reply:
+1